lib/msf/core/auxiliary/password_cracker.rb

```diff
@@ -75,16 +75,28 @@ def john_lm_upper_to_ntlm(pwd, hash)
   #
   # @return [nilClass] if there is no active framework db connection
   # @return [Metasploit::Framework::PasswordCracker::Cracker] if it successfully creates a Password Cracker object
-  def new_password_cracker
-    return nil unless framework.db.active
-    Metasploit::Framework::PasswordCracker::Cracker.new(
+  def new_password_cracker(cracking_application)
+    fail_with(Msf::Module::Failure::BadConfig, "Password cracking is not available without an active data database connection.") unless framework.db.active
```

Comment:
I wasn't sure if this was intentional or not:

```suggestion
    fail_with(Msf::Module::Failure::BadConfig, "Password cracking is not available without an active database connection.") unless framework.db.active
```

Reply:
Good catch!... Fixed.